tests/vm_test.go

```diff
@@ -1546,9 +1546,11 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 					Expect(err).ToNot(HaveOccurred(), "Cannot generate VMs manifest")
 
 					By("Checking VM creation permission using k8s client binary")
-					stdOut, _, err := tests.RunCommand(k8sClient, "auth", "can-i", "create", "vms", "--as", testUser)
-					Expect(err).ToNot(HaveOccurred())
-					Expect(strings.TrimSpace(stdOut)).To(Equal("yes"))
+					// It might take time for the role to propagate
+					Eventually(func() string {
+						stdOut, _, _ := tests.RunCommand(k8sClient, "auth", "can-i", "create", "vms", "--as", testUser)
+						return strings.TrimSpace(stdOut)
+					}, 120*time.Second, 10*time.Second).Should(Equal("yes"), fmt.Sprintf("test account '%s' was never granted permission to create a VM", testUser))
```

Comment:
I would expect that we retry each second and wait up to 10 seconds. I am not so fluent in this area: What 3xactly need time to be propagated?

Reply:
The BeforeEach of this test is essentially running "oc adm policy add-role-to-user admin testuser". This does not appear to take effect instantly.

Your approach to timing is probably more reasonable.